lld/COFF/Writer.cpp

```diff
@@ -2555,6 +2566,33 @@ void Writer::addBaserelBlocks(std::vector<Baserel> &v) {
   relocSec->addChunk(make<BaserelChunk>(page, &v[i], &v[0] + j));
 }
 
+void Writer::createDynamicRelocs() {
+  if (!ctx.dynamicRelocs)
+    return;
+
+  const uint32_t coffHeaderOffset = dosStubSize + sizeof(PEMagic);
+  const uint32_t peHeaderOffset = coffHeaderOffset + sizeof(coff_file_header);
+  const uint32_t dataDirOffset = peHeaderOffset + sizeof(pe32plus_header);
+
+  // Adjust the Machine field in the COFF header to AMD64.
+  ctx.dynamicRelocs->add(IMAGE_DVRT_ARM64X_FIXUP_TYPE_VALUE, sizeof(uint16_t),
+                         coffHeaderOffset + offsetof(coff_file_header, Machine),
+                         AMD64);
+
+  // Adjust the load config directory.
```

Comment:
"Adjust", as in setting the values to 0, if I understand the code correctly? Can you adjust (no pun intended ;-) ) the comment to be more clear about what adjustment it does?

Then again, I guess that the FIXME below means that clearing it isn't the final intended behaviour, but an initial stepping stone?

Reply:
Ultimately, on ARM64X, we should define two separate load configs: native and EC. The native load config is written to the header, and ARM64X relocations "adjust" it to reference the EC load config. If the EC load config is missing, as in the test from this PR, MSVC emits a warning (this part is not yet implemented) and resets the header value so that the EC view has no load config.

Testing and supporting two load configs isn’t currently possible in lld-link due to duplicated `_load_config_used` symbols. We need support for separated namespaces first.